core/model/commands/transfer.hpp

```diff
@@ -15,56 +15,61 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-#ifndef CORE_DOMAIN_TRANSACTIONS_TRANSFERTRANSACTION_HPP_
-#define CORE_DOMAIN_TRANSACTIONS_TRANSFERTRANSACTION_HPP_
+#ifndef CORE_DOMAIN_TRANSFER_HPP_
+#define CORE_DOMAIN_TRANSFER_HPP_
 
-#include "../transaction.hpp"
-#include "command.hpp"
-#include "../objects/object.hpp"
+#include "../objects/domain.hpp"
+#include "../objects/asset.hpp"
+#include "../objects/message.hpp"
 
 #include <string>
-#include <type_traits>
-
+#include <iostream>
 #include "../../service/json_parse.hpp"
 
 namespace command {
 
-    template<typename T,
-        std::enable_if_t<
-            std::is_base_of<AbsObject, T>::value,std::nullptr_t
-        > = nullptr
-    >
-    class Transfer : public Command{
-        T object;
-    public:
-        Transfer(T object):
-            object(object)
-        {}
-
-        std::string getAsJson() const{
-            object.getAsJson();
-        }
-
-        std::string getCommandName() const{
-            return "Transfer";
-        }
-
-        using Object = json_parse::Object;
-        using Rule = json_parse::Rule;
-        using Type = json_parse::Type;
-
-        Object dump() {
-            Object obj = Object(Type::DICT);
-            obj.dictSub.insert( std::make_pair( "command", Object(Type::STR, getCommandName())));
-            obj.dictSub.insert( std::make_pair( "object", object.dump()));
-            return obj;
-        }
-
-        static Rule getJsonParseRule() {
-            Rule obj = Rule(Type::DICT);
-            obj.dictSub.insert( std::make_pair( "command", Rule(Type::STR)));
-            obj.dictSub.insert( std::make_pair( "object", T::getJsonParseRule()));
-            return obj;
+template <typename T>
+class Transfer: public T {
+
+   std::string senderPublicKey;
+   std::string receiverPublicKey;
+  public:
+
+    Transfer(
+        std::string senderPubkey,
+        std::string receiverPubkey,
+        std::string name,
+        int value
+    );
+
+    std::string getCommandName() const{
+        return "Transfer";
+    }
+
+    using Rule = json_parse::Rule;
+    using Type = json_parse::Type;
+    using Object = json_parse::Object;
+
+    Transfer(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'Transfer' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4478583006/issues/source?bid=3712491&fileBranchId=3923843#l53)